js/src/forum/index.js

```diff
@@ -48,35 +49,57 @@ function makeShortcut(id, key, editorDriver) {
   };
 }
 
+function markdownToolbarItems(oldFunc) {
+  const items = typeof oldFunc === 'function' ? oldFunc() : new ItemList();
+
+  function tooltip(name, hotkey) {
+    return app.translator.trans(`flarum-markdown.forum.composer.${name}_tooltip`) + (hotkey ? ` <${modifierKey}-${hotkey}>` : '');
+  }
+
+  const makeApplyStyle = (id) => {
+    return () => applyStyle(id, this.attrs.composer.editor);
+  };
+
+  items.add('header', <MarkdownButton title={tooltip('header')} icon="fas fa-heading" onclick={makeApplyStyle('header')} />, 1000);
+  items.add('bold', <MarkdownButton title={tooltip('bold', 'b')} icon="fas fa-bold" onclick={makeApplyStyle('bold')} />, 900);
+  items.add('italic', <MarkdownButton title={tooltip('italic', 'i')} icon="fas fa-italic" onclick={makeApplyStyle('italic')} />, 800);
+  items.add(
+    'strikethrough',
+    <MarkdownButton title={tooltip('strikethrough')} icon="fas fa-strikethrough" onclick={makeApplyStyle('strikethrough')} />,
+    700
+  );
+  items.add('quote', <MarkdownButton title={tooltip('quote')} icon="fas fa-quote-left" onclick={makeApplyStyle('quote')} />, 600);
+  items.add('spoiler', <MarkdownButton title={tooltip('spoiler')} icon="fas fa-exclamation-triangle" onclick={makeApplyStyle('spoiler')} />, 500);
+  items.add('code', <MarkdownButton title={tooltip('code')} icon="fas fa-code" onclick={makeApplyStyle('code')} />, 400);
+  items.add('link', <MarkdownButton title={tooltip('link')} icon="fas fa-link" onclick={makeApplyStyle('link')} />, 300);
+  items.add('image', <MarkdownButton title={tooltip('image')} icon="fas fa-image" onclick={makeApplyStyle('image')} />, 200);
+  items.add(
+    'unordered_list',
+    <MarkdownButton title={tooltip('unordered_list')} icon="fas fa-list-ul" onclick={makeApplyStyle('unordered_list')} />,
+    100
+  );
+  items.add('ordered_list', <MarkdownButton title={tooltip('ordered_list')} icon="fas fa-list-ol" onclick={makeApplyStyle('ordered_list')} />, 0);
+
+  return items;
+}
+
 app.initializers.add('flarum-markdown', function (app) {
   extend(BasicEditorDriver.prototype, 'keyHandlers', function (items) {
     items.add('bold', makeShortcut('bold', 'b', this));
     items.add('italic', makeShortcut('italic', 'i', this));
   });
 
-  extend(TextEditor.prototype, 'toolbarItems', function (items) {
-    const tooltip = (name, hotkey) => {
-      return app.translator.trans(`flarum-markdown.forum.composer.${name}_tooltip`) + (hotkey ? ` <${modifierKey}-${hotkey}>` : '');
-    };
-
-    const makeApplyStyle = (id) => {
-      return () => applyStyle(id, this.attrs.composer.editor);
-    };
+  if (TextEditor.prototype.markdownToolbarItems) {
+    override(TextEditor.prototype, 'markdownToolbarItems', markdownToolbarItems);
+  } else {
+    TextEditor.prototype.markdownToolbarItems = markdownToolbarItems;
+  }
```

Comment:
Shouldn't we be able to `override` in either case?

Reply:
When I tried that, it seemed to error at runtime.

Reply:
![image](https://user-images.githubusercontent.com/7406822/147704200-969755ff-4f15-40c7-aeaa-f8a6b7fdc34e.png)

Reply:
Hmm, we might want to change that in core. But makes sense for now.

Reply:
I agree